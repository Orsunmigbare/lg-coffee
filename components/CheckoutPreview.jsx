import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Loading from './Loading';
import { formatUSD } from '../utils/formatUSD';

const CheckoutPreview = ({ cart }) => {
  if (!cart) <Loading data={cart} />;

  const [cartInStorage, setCartInStorage] = useState(cart);
  const [isCartEmpty, setCartToEmpty] = useState();

  useEffect(() => {
    // eslint-disable-next-line no-unused-expressions
    cartInStorage ? setCartToEmpty(false) : setCartToEmpty(true);
  }, [cartInStorage]);

  const handleRemoveFromCart = (id) => {
    const updatedCart = cartInStorage;

    delete updatedCart[id];

    setCartInStorage(updatedCart);

    if (Object.keys(updatedCart).length === 0) {
      localStorage.removeItem('cart');
      setCartToEmpty(true);
    } else localStorage.setItem('cart', JSON.stringify(cartInStorage));
  };

  return (
    <>
      <Container fluid>
        {cartInStorage === undefined ||
        cartInStorage === null ||
        isCartEmpty ? (
          <h2 className="text-light">No Products</h2>
        ) : (
          <>
            <Row>
              <h1 className="text-light mt-5 mb-4">Checkout</h1>
            </Row>
            <Row className="d-flex align-items-center border-light">
              <h2 className="text-light ellipse">Products</h2>
            </Row>

            {Object.keys(cartInStorage).map((i) => (
              <a
                href={cartInStorage[i].sku}
                style={{ textDecoration: 'none' }}
                key={cartInStorage[i].sku}
              >
                <Row className="d-flex align-items-center rounded bg-dark">
                  <Col xs={4}>
                    <img
                      src={cartInStorage[i].images[0]}
                      alt={cartInStorage[i].name}
                      style={{ height: '100px', width: '100px' }}
                    />
                  </Col>
                  <Col xs={8} className="d-flex flex-column">
                    <Row className="align-self-end mr-2">
                      <h4
                        style={{ width: '100%', margin: '0' }}
                        className="text-light ellipse"
                      >
                        {cartInStorage[i].name}
                      </h4>
                    </Row>
                    <Row className="align-self-end align-items-end mr-2">
                      <p
                        className="text-light mt-2"
                        style={{ margin: '0', padding: '0' }}
                      >
                        ${formatUSD(cartInStorage[i].price)} USD
                      </p>
                      <p
                        className="text-light ml-4 mr-2"
                        style={{ margin: '0', padding: '0' }}
                      >
                        Qty: {cartInStorage[i].qty}
                      </p>
                      <button
                        type="button"
                        style={{
                          border: 'none',
                          background: 'none',
                          margin: '0',
                          padding: '0',
                        }}
                        className="btn btn-outline-light font-weight-bolder ml-4"
                        onClick={() => {
                          handleRemoveFromCart(i);
                        }}
                      >
                        X
                      </button>
                    </Row>
                  </Col>
                </Row>
              </a>
            ))}
          </>
        )}
      </Container>
    </>
  );
};

CheckoutPreview.propTypes = {
  // eslint-disable-next-line react/require-default-props
  cart: PropTypes.shape({
    images: PropTypes.string,
    name: PropTypes.string,
    price: PropTypes.string,
    qty: PropTypes.string,
  }),
};

export default CheckoutPreview;
