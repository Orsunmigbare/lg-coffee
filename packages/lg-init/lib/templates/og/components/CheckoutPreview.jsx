import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Loading from './Loading';
import { formatUSD } from '../utils/formatUSD';
import { H2, H3, H4 } from './_helpers';

const CheckoutPreview = ({ cart }) => {
  if (!cart) <Loading data={cart} />;

  const [cartInStorage, setCartInStorage] = useState(cart);
  const [isCartEmpty, setCartToEmpty] = useState();

  useEffect(() => {
    // eslint-disable-next-line no-unused-expressions
    cartInStorage ? setCartToEmpty(false) : setCartToEmpty(true);
    
  }, [cartInStorage]);

  const handleRemoveFromCart = (id) => {
    const updatedCart = cartInStorage;

    delete updatedCart[id];

    setCartInStorage(updatedCart);

    if (Object.keys(updatedCart).length === 0) {
      localStorage.removeItem('cart');
      setCartToEmpty(true);
    } else localStorage.setItem('cart', JSON.stringify(cartInStorage));
  };

  return (
    <>
      <Container>
        {cartInStorage === undefined ||
        cartInStorage === null ||
        isCartEmpty ? (
          <H2 className="m-4">No Products</H2>
        ) : (
          <>
            <Row className="d-flex align-items-center">
              <H3 className="ellipse pb-5 pl-5">In Cart</H3>
            </Row>

            {Object.keys(cartInStorage).map((i) => (
              <Row
                key={cartInStorage[i].sku}
                className="d-flex align-items-center rounded px-5"
              >
                <Col xs={4}>
                  <img
                    src={cartInStorage[i].images[0]}
                    alt={cartInStorage[i].name}
                    style={{ height: '100px', width: '100px' }}
                  />
                </Col>
                <Col xs={8} className="d-flex flex-column">
                  <Row className="align-self-end mr-2">
                    <H4
                      style={{ width: '100%', margin: '0' }}
                      className="ellipse"
                    >
                      {cartInStorage[i].name}
                    </H4>
                  </Row>
                  <Row className="align-self-end align-items-end mr-2">
                    <p className="mt-2" style={{ margin: '0', padding: '0' }}>
                      ${formatUSD(cartInStorage[i].price)} USD
                    </p>
                    <p
                      className="ml-4 mr-2"
                      style={{ margin: '0', padding: '0' }}
                    >
                      Qty: {cartInStorage[i].qty}
                    </p>
                    <button
                      type="button"
                      style={{
                        border: 'none',
                        background: 'none',
                        margin: '0',
                        padding: '0',
                      }}
                      className="btn btn-outline-danger font-weight-bolder ml-4"
                      onClick={() => {
                        handleRemoveFromCart(i);
                      }}
                    >
                      X
                    </button>
                  </Row>
                </Col>
              </Row>
            ))}
          </>
        )}
      </Container>
    </>
  );
};

CheckoutPreview.propTypes = {
  // eslint-disable-next-line react/require-default-props
  cart: PropTypes.shape({
    images: PropTypes.string,
    name: PropTypes.string,
    price: PropTypes.string,
    qty: PropTypes.string,
  }),
};

export default CheckoutPreview;
