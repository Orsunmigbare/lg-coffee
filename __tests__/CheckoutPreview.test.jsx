import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CheckoutPreview from '../components/CheckoutPreview';

const cart = {
  product: {
    productTitle: 'product',
    productDescription: 'this is an awesome product you should buy 2',
    productPrice: '$2.87',
    productSku: 'abc_123',
    productImages: [
      '/path_01.png',
      '/path_02.png',
      '/path_03.png',
    ],
    qty: 12,
  },
};

it('CheckoutPreview component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CheckoutPreview cart={cart} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('CheckoutPreview component renders unchanged', () => {
  const tree = renderer.create(<CheckoutPreview cart={cart} />).toJSON();
  expect(tree).toMatchSnapshot();
});
