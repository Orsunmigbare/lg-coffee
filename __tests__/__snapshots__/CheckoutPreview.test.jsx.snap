// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPreview component renders unchanged 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <h1
      className="text-light mt-5 mb-4"
    >
      Checkout
    </h1>
  </div>
  <div
    className="d-flex align-items-center border-light row"
  >
    <div
      className="col-8"
    >
      <h2
        className="text-light ellipse"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        Product
      </h2>
    </div>
    <div
      className="col-3"
    >
      <h2
        className="text-light"
      >
        Price
      </h2>
    </div>
    <div
      className="col-1"
    >
      <h2
        className="text-light"
      >
        Qty
      </h2>
    </div>
  </div>
  <div
    className="d-flex align-items-center border-light row"
  >
    <div
      className="col-3"
    >
      <img
        alt="product"
        src="/path_01.png"
        style={
          Object {
            "height": "100px",
            "width": "100px",
          }
        }
      />
    </div>
    <div
      className="col-5"
    >
      <h4
        className="text-light ellipse"
        style={
          Object {
            "margin": "0",
            "width": "100%",
          }
        }
      >
        product
      </h4>
    </div>
    <div
      className="col-3"
    >
      <p
        className="text-light"
        style={
          Object {
            "margin": "0",
          }
        }
      >
        $2.87
      </p>
    </div>
    <div
      className="d-flex align-items-center justify-content-between col-1"
    >
      <p
        className="text-light"
        style={
          Object {
            "margin": "0",
          }
        }
      >
        12
      </p>
      <button
        className="btn btn-outline-light font-weight-bolder ml-4"
        onClick={[Function]}
        style={
          Object {
            "background": "none",
            "border": "none",
            "margin": "0",
            "padding": "0",
          }
        }
        type="button"
      >
        X
      </button>
    </div>
  </div>
</div>
`;
